src/test/java/uk/nhs/digital/nhsconnect/lab/results/inbound/RecepProducerServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package uk.nhs.digital.nhsconnect.lab.results.inbound;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.HealthAuthorityNameAndAddress;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Interchange;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.InterchangeHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.InterchangeTrailer;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageHeader;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class RecepProducerServiceTest {
```

Comment:
Are there any 'unhappy' paths we could cover?

Reply:
Yes, we can trigger an `EdifactValidationException` if we give empty strings for some values.